packages/react-ui/src/app/builder/step-settings/piece-settings/connection-select.tsx

```diff
@@ -40,41 +47,87 @@ const ConnectionSelect = React.memo((params: ConnectionSelectProps) => {
     pieceName: params.piece.name,
     cursor: undefined,
     limit: 100,
-    projectId: authenticationSession.getProjectId(),
+    projectId: authenticationSession.getProjectId() ?? '',
   });
-
-  const addBrackets = (str: string) => `{{connections['${str}']}}`;
+  const currentlySelectedConnection = useMemo(() => {
```

Comment:
You no longer need to traverse the whole connections page, you just need the name :") and it's already in form value.

Reply:
I need it to know the connection type for the oauth dialog.